### DISCUSSION TITLE
Automating Daily Price Updates in Supabase with Web Scraping: Seeking Guidance

### DISCUSSION THREAD

**Discussion by cperezdearce1** [supabase/supabase#22199] [Questions]
Objective
I'm looking to automate the process of updating product prices in a Supabase table from an e-commerce website. My goal is to scrape the current price of each product, identified by its ID, and store this information along with a timestamp in a database, with updates occurring daily at 12 pm.

Current Setup
Database: I have a Supabase table containing a list of product IDs.
Scraping Tool: I'm using Puppeteer to scrape the product prices from the website. This setup works well locally in Visual Studio, successfully retrieving each product's price.

Challenge
I need to transition from manually running the script to having it execute automatically every day, without relying on my local machine. Additionally, it's crucial for the process to update the Supabase table with the new prices and the time of the update.

Considerations
Automated Execution: I'm considering using Supabase Edge Functions or Google Cloud Functions for automation, though I'm open to other suggestions. The key requirements are daily automatic execution and database updating.
Experience Level: While I'm not a developer by trade, I'm keen on learning and can pick up new concepts quickly, especially with some guidance on getting started.

Request for Guidance
Could you advise on the best approach to automate this scraping and updating process? Any basic guidance or resources for setting up the chosen solution would be greatly appreciated, as I'm ready to dive into more detailed learning once I have a clear starting point.

**Comment by GaryAustin1** [comment]
Depending on the complexity of the scraping either an in database function or an edge function could do the fetching of data and updating of the database.  You could call either function with the pg_cron extension and either a database function call or a http/pg_net extension call of your edge function.  There are docs in supabase on those extensions and plenty of cron examples on the web.

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/database/extensions/pg_cron#invoke-supabase-edge-function-every-minute

**Reply by cperezdearce1** [reply]
Hi @GaryAustin1 , thank so much for your response. To be honest, I can't achieve what I was intending to do. I was using node.js to code the puppeteer scrapper, but I don't know how to implement a function like that in a supabase edge function. Any guide that I could use to learn 
how to do it?

**Reply by sSKEAMm** [reply]
hey man did you even get a chance to finish this project?
